src/server/pfs/server/driver.go

```diff
@@ -25,14 +26,56 @@ import (
 	"github.com/pachyderm/pachyderm/src/server/pkg/watch"
 
 	etcd "github.com/coreos/etcd/clientv3"
+	"github.com/gogo/protobuf/proto"
 	"github.com/gogo/protobuf/types"
 	"github.com/hashicorp/golang-lru"
 	"google.golang.org/grpc"
 )
 
+// ValidateRepoName determines if a repo name is valid
+func ValidateRepoName(name string) error {
+	match, _ := regexp.MatchString("^[a-zA-Z0-9_-]+$", name)
+
+	if !match {
+		return fmt.Errorf("repo name (%v) invalid: only alphanumeric characters, underscores, and dashes are allowed", name)
+	}
+
+	return nil
+}
+
+// ListFileMode specifies how ListFile executes.
+type ListFileMode int
+
+const (
+	// ListFileNORMAL computes sizes for files but not for directories
+	ListFileNORMAL ListFileMode = iota
+	// ListFileFAST does not compute sizes for files or directories
+	ListFileFAST
+	// ListFileRECURSE computes sizes for files and directories
+	ListFileRECURSE
+)
+
+// IsPermissionError returns true if a given error is a permission error.
+func IsPermissionError(err error) bool {
+	return strings.Contains(err.Error(), "has already finished")
+}
+
+// CommitEvent is an event that contains a CommitInfo or an error
+type CommitEvent struct {
+	Err   error
+	Value *pfs.CommitInfo
+}
+
+// CommitStream is a stream of CommitInfos
+type CommitStream interface {
+	Stream() <-chan CommitEvent
+	Close()
+}
+
 type collectionFactory func(string) col.Collection
 
-type driver struct {
+// Driver implements the underlying pfs semantics.
+type Driver struct {
```

Comment:
I can, I'll have to rejigger how it gets instantiated a bit but shouldn't be too hard.